tgui/packages/tgui/interfaces/AntagInfoHeretic.tsx

```diff
@@ -219,63 +224,65 @@ const InformationSection = (props) => {
   );
 };
 
-const ResearchedKnowledge = (props) => {
-  const { data } = useBackend<KnowledgeInfo>();
-  const { learnedKnowledge } = data;
-
-  return (
-    <Stack.Item grow>
-      <Section title="Researched Knowledge" fill scrollable>
-        <Stack vertical>
-          {(!learnedKnowledge.length && 'None!') ||
-            learnedKnowledge.map((learned) => (
-              <Stack.Item key={learned.name}>
-                <Button
-                  width="100%"
-                  color={learned.color}
-                  content={`${learned.hereticPath} - ${learned.name}`}
-                  tooltip={learned.desc}
-                />
-              </Stack.Item>
-            ))}
-        </Stack>
-      </Section>
-    </Stack.Item>
-  );
-};
-
-const KnowledgeShop = (props) => {
+const KnowledgeTree = (props) => {
   const { data, act } = useBackend<KnowledgeInfo>();
-  const { learnableKnowledge } = data;
+  const { knowledge_tiers } = data;
 
   return (
-    <Stack.Item grow>
-      <Section title="Potential Knowledge" fill scrollable>
-        {(!learnableKnowledge.length && 'None!') ||
-          learnableKnowledge.map((toLearn) => (
-            <Stack.Item key={toLearn.name} mb={1}>
-              <Button
-                width="100%"
-                color={toLearn.color}
-                disabled={toLearn.disabled}
-                content={`${toLearn.hereticPath} - ${
-                  toLearn.cost > 0
-                    ? `${toLearn.name}: ${toLearn.cost}
-                  point${toLearn.cost !== 1 ? 's' : ''}`
-                    : toLearn.name
-                }`}
-                tooltip={toLearn.desc}
-                onClick={() => act('research', { path: toLearn.path })}
-              />
-              {!!toLearn.gainFlavor && (
-                <BlockQuote>
-                  <i>{toLearn.gainFlavor}</i>
-                </BlockQuote>
-              )}
+    <Section title="Research Tree" fill scrollable>
+      <Stack vertical>
+        {(!knowledge_tiers.length && 'None!') ||
+          knowledge_tiers.map((tier, i) => (
+            <Stack.Item key={i}>
+              <Flex
+                direction="row"
+                justify="center"
+                align="center"
+                backgroundColor="#303030"
+                wrap="wrap"
+              >
+                {
+                  tier.nodes.map((node) => (
```

Comment:
`{tier.nodes.map((node) => (`
🤔 are you using prettier? this should be fixing itself on save